gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/task/EhrExtractMessageHandler.java

```diff
@@ -279,4 +303,73 @@ private String parseToOdsCode(RCMRIN030000UK06Message payload) {
     private String parseMessageRef(RCMRIN030000UK06Message payload) {
         return payload.getId().getRoot();
     }
+
+    private String parseFilename(String description) throws ParseException {
+        Pattern pattern = Pattern.compile("Filename=\"([A-Za-z\\d\\-_. ]*)\"");
+        Matcher matcher = pattern.matcher(description);
+
+        if (matcher.find()) {
+            return matcher.group(1);
+        }
+        throw new ParseException("Unable to parse originalFilename", 0);
+    }
+
+    private String parseContentType(String description) throws ParseException {
+        Pattern pattern = Pattern.compile("ContentType=([A-Za-z\\d\\-/]*)");
+        Matcher matcher = pattern.matcher(description);
+
+        if (matcher.find()) {
+            return matcher.group(1);
+        }
+        throw new ParseException("Unable to parse ContentType", 0);
+    }
+
+    private boolean parseIsCompressed(String description) throws ParseException {
```

Comment:
There are few similar methods to this one, just having a different pattern to be matched. Maybe it would make sense to create a common method that can be called in each of those.